ProjectName/src/screens/PaymentDetails/PaymentDetailsForm/PaymentDetailsForm.tsx

```diff
@@ -0,0 +1,83 @@
+import {useNavigation} from '@react-navigation/native';
+import React from 'react';
+import {useForm, SubmitHandler} from 'react-hook-form';
+import {Text, View, Pressable} from 'react-native';
+
+import CustomAmountInput from './components/customAmountInput/CustomAmountInput';
+import CustomButton from './components/customButton/CustomButton';
+import CustomInput from './components/customInput/CustomInput';
+import CustomMessageInput from './components/customMessageInput/CustomMessageInput';
+import {styles} from './payment-details-form.styles';
+import {PaymentDetailsFormData} from './payment-details-form.types';
+import {NavigationType} from '../../../navigation/navigation.types';
+
+const PaymentDetailsForm: React.FC = () => {
+  const navigation = useNavigation<NavigationType>();
+
+  const {
+    control,
+    handleSubmit,
+    formState: {errors},
+  } = useForm<PaymentDetailsFormData>({
```

Comment:
It would be nice to have some validation for fiels.
I would suggest using `resolver`. You may pick one which you like the most (give a try `zod` for example),